src/pat-redactor.js

```diff
@@ -1,121 +1,191 @@
-/**
- * Pattern for Redactor
- * Copyright 2013-2014 Simplon B.V. - Wichert Akkerman
+/*
+ * Pattern for Redactor 3
+ * Copyright 2015-2016 Syslab.com Gmbh
  */
+(function(root, factory) {
+  if (typeof define === "function" && define.amd) {
+    // AMD. Register as an anonymous module.
+    define([
+      "jquery",
+      "underscore",
+      "pat-base",
+      "pat-parser",
+      "redactor",
+      "redactor-alignment",
+      //"redactor-clips",
+      "redactor-counter",
+      "redactor-definedlinks",
+      "redactor-filemanager",
+      "redactor-fullscreen",
+      "redactor-imagemanager",
+      "redactor-inlinestyle",
+      "redactor-limiter",
+      "redactor-properties",
+      "redactor-table",
+      //"redactor-textdirection",
+      //"redactor-textexpander",
+      "redactor-video"
+    ], function($, _, Base, Parser) {
+      return factory.apply(this, arguments);
+    });
+  } else {
+    factory(
+      root.jQuery,
+      _,
+      root.patterns,
+      root.patterns.Base,
+      root.patterns.Parser
+    );
+  }
+})(this, function($, _, Base, Parser) {
+  var parser = new Parser("redactor");
 
-(function (root, factory) {
-    if (typeof define === 'function' && define.amd) {
-        // AMD. Register as an anonymous module.
-        define([
-            "jquery",
-            "underscore",
-            "pat-registry",
-            "pat-utils",
-            "pat-parser",
-            "pat-pluggable",
-            "redactor",
-            ], function ($, _, registry, utils, Parser, pluggablePattern, Redactor) {
-                return factory.apply(this, arguments);
-        });
-    } else {
-        factory(root.jQuery, _, root.patterns, utils, root.patterns.Parser, pluggablePattern, Redactor);
-    }
-}(this, function($, _, registry, utils, Parser, pluggablePattern, Redactor) {
-    var parser = new Parser('redactor');
+  parser.add_argument("toolbar-type", "standard", ["standard", "fixed", "air"]);
+  parser.add_argument("toolbar-external", null);
+  parser.add_argument("toolbar-fixed-target", null);
+
+  parser.add_argument(
+    "buttons",
+    [
+      "html",
+      "undo",
+      "redo",
+      "format",
+      "bold",
+      "italic",
+      "deleted",
+      "lists",
+      "link",
+      "line",
+      "image"
+    ],
+    [
+      "bold",
+      "deleted",
+      "file",
+      "format",
+      "html",
+      "image",
+      "indent",
+      "italic",
+      "line",
+      "link",
+      "lists",
+      "ol",
+      "outdent",
+      "redo",
+      "sub",
+      "sup",
+      "ul",
+      "underline",
+      "undo"
+    ],
+    true
+  );
+
+  parser.add_argument(
+    "formatting",
+    ["p", "blockquote", "pre", "h1", "h2", "h3", "h4", "h5"],
+    ["p", "blockquote", "pre", "h1", "h2", "h3", "h4", "h5"],
+    true
+  );
+
+  parser.add_argument(
+    "plugins",
```

Comment:
No plugins for: bufferbuttions (undo and redo are normal buttons), source (now an option + button), codemirror (now an option for source option), romanlisting

Reply:
we must port our own romanlisting plugin before updating ikath!